app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardViewModel.kt

```diff
@@ -69,5 +78,39 @@ class DashboardViewModel(
             .addTo(compositeDisposable)
     }
 
+    fun getPinUpdate() {
+        val prefData = appRepository.getPrefCountry()
+        if (prefData == null) {
+            _pinData.value = null
+        } else {
+            appRepository
+                .confirmed()
+                .map { stream -> stream.firstOrNull() { if (it.provinceState != null) it.provinceState == prefData.provinceState else it.countryRegion == prefData.countryRegion } }
```

Comment:
You can add `.observeOn(schedulerProvider.io())` above the map function, so the map process running on `schedulerProvider.io` thread explicitly

Reply:
looks great, done